api.md

```diff
@@ -919,6 +922,82 @@ The expected output:
 
 ---
 
+## interpolate() [](interpolate)
+
+The `interpolate` function does linear interpolation for missing values.
+It can only be used in an aggregation query with [time_bucket_gapfill](#time_bucket_gapfill).
+The `interpolate` function call cannot be nested inside other function calls.
+
+#### Required Arguments [](interpolate-required-arguments)
+
+|Name|Description|
+|---|---|
+| `value` | The value to interpolate (int2/int4/int8/float4/float8) |
+
+#### Optional Arguments [](interpolate-optional-arguments)
+
+|Name|Description|
+|---|---|
+| `prev` | The lookup expression for values before the gapfill time range (record) |
+| `next` | The lookup expression for values after the gapfill time range (record) |
+
+The `prev` and `next` expression will only be evaluated when no suitable value is returned by the outer query.
+The returned record for `prev` and `next` needs to be a time, value tuple.
+The datatype of time needs to be the same as the time datatype in the `time_bucket_gapfill` call.
+The datatype of value needs to be the same as the `value` datatype of the `interpolate` call.
+
+#### Sample Usage [](interpolate-examples)
+
+Get the temperature every day for each device over the last week interpolating for missing readings:
+```sql
+SELECT
+  time_bucket_gapfill('1 day', time, now() - interval '1 week', now()) as time,
+  device_id,
+  avg(temperature) AS value,
+  interpolate(avg(temperature))
+FROM metrics
+WHERE time > now () - interval '1 week'
+GROUP BY 1,2
+ORDER BY 1 DESC;
+
+          time          | device_id | value | interpolate
+------------------------+-----------+-------+-------------
+ 2019-01-10 01:00:00+01 |         1 |       |
+ 2019-01-11 01:00:00+01 |         1 |   5.0 |         5.0
+ 2019-01-12 01:00:00+01 |         1 |       |         6.0
+ 2019-01-13 01:00:00+01 |         1 |   7.0 |         7.0
+ 2019-01-14 01:00:00+01 |         1 |       |         6.5
+ 2019-01-15 01:00:00+01 |         1 |   8.0 |         8.0
+ 2019-01-16 01:00:00+01 |         1 |   9.0 |         9.0
+(7 row)
+```
+
+Get the average temperature every day for each device over the last 7 days interpolating for missing readings with lookup queries for values before and after the gapfill time range:
+```sql
+SELECT
+  time_bucket_gapfill('1 day', time, now() - interval '1 week', now()) as time,
+  device_id,
+  interpolate(avg(temperature) AS value,
+    (SELECT (time,temperature) FROM metrics m2 WHERE m2.time < now() - interval '1 week' AND m.device_id = m2.device_id),
+    (SELECT (time,temperature) FROM metrics m2 WHERE m2.time > now() AND m.device_id = m2.device_id)
+  )
+FROM metrics
+WHERE time > now () - interval '1 week'
+GROUP BY 1,2
```

Comment:
I don't think "GROUP BY 1, 2 ORDER BY 1" is less clear to users as column names.  Can we change this (and all below) to 

"GROUP BY time, device_d ORDER BY time" ?

(I noticed below that you responded:  "I think the numerical syntax is superior because otherwise you have to repeat the expression or use an alias which might give unexpected results if you have columns by the same name. The current version of the document also uses that syntax."  Maybe we can discuss with team if we have overall preference with docs.)